components/availability/src/Availability.svelte

```diff
@@ -509,6 +526,66 @@
     }
     return freeBusyCalendars;
   }
+
+  // Figure out if a given TimeSlot is the first one in a pending, or selected, block.
+  function getBlockTimes(slot: SelectableSlot, day: Day) {
+    let slotIndex = day.slots.indexOf(slot);
+    let wrappingSortedSlot = sortedSlots.find(
+      (block) => block.start_time === slot.start_time,
+    );
+    // A slot is first in a Pending block if:
+    // it's pending
+    // and it's not busy
+    // and either:
+    // the slot immediately before it isn't pending
+    // OR
+    // the slot immediately before it is busy
+    if (
+      slot.selectionPending &&
+      slot.availability !== AvailabilityStatus.BUSY &&
+      (!day.slots[slotIndex - 1]?.selectionPending ||
+        day.slots[slotIndex - 1]?.availability === AvailabilityStatus.BUSY)
+    ) {
+      let pendingEndTime =
+        day.slots.find((s) => {
```

Comment:
can `s` please be changed to a variable name that is readable? Maybe `pendingSlot`?